java/vector/src/main/java/org/apache/arrow/vector/complex/NonNullableStructVector.java

```diff
@@ -320,6 +322,20 @@ public int hashCode(int index, ArrowBufHasher hasher) {
     return visitor.visit(this, value);
   }
 
+  @Override
+  public void validate() {
+    if (getValueCount() < 0) {
+      throw new RuntimeException("vector valueCount is negative");
+    }
+
+    if (getNullCount() > getValueCount()) {
```

Comment:
For the Java implementation, since the null count is computed each time, we shoud never expect getNullCount() > getValueCount()?

Reply:
Thanks, fixed.